highlighters/main/main-highlighter.zsh

```diff
@@ -464,35 +535,15 @@ _zsh_highlight_highlighter_main_paint()
      style=reserved-word # de facto a reserved word, although not de jure
      next_word=':start:'
    elif [[ $this_word == *':start:'* ]] && (( in_redirection == 0 )); then # $arg is the command word
-     if [[ -n ${(M)ZSH_HIGHLIGHT_TOKENS_PRECOMMANDS:#"$arg"} ]]; then
-      style=precommand
-     elif [[ "$arg" = "sudo" ]] && { _zsh_highlight_main__type sudo; [[ -n $REPLY && $REPLY != "none" ]] }; then
+     if (( ${+precommand_options[$arg]} )) && { _zsh_highlight_main__type $arg; [[ -n $REPLY && $REPLY != "none" ]] }; then
```

Comment:
`{ _zsh_highlight_main__type $arg; [[ -n $REPLY && $REPLY != "none" ]] }` is non obvious in purpose. Perhaps a comment should be added (or have __type return 1 if the command doesn't exist). Prior issue; doesn't block merging.

Reply:
I find `if {f; [[ -n $REPLY ]]}; then` idiomatic, especially with the braces. What would a comment say?

s/;/&&/ etc sounds good.

Reply:
It took me a few seconds to realize that the point of the code was checking for the command existing on this system is all. Not attached to adding a comment if you don't think it necessary,

Reply:
Good idea.

Reply:
Done